### PR TITLE
OSSM-4194 Made change to the host in the DestinationRule v2.4

### PR DESCRIPTION
**Pull request by mayleighnmyers**
https://issues.redhat.com/browse/OSSM-4194

- changed `"*.local"` to `".{{ .Values.global.proxy.clusterDomain }}"`
- Ran `make gen` and `make lint`

### CODE REVIEW

File: resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ metadata:
     app: istio
     release: {{ .Release.Name }}
 spec:
-  host: "*.local"
+  host: "*.clusterDomain"
```
Comment (luksa) [inline resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml]: No. This should be something like `*.{{ .Values.global.proxy.clusterDomain }}`.

Comment (mayleighnmyers) [reply resources/helm/overlays/mesh-config/templates/enable-mesh-mtls.yaml]: Made the change, both make gen and make lint passed on my side!

**Comment by luksa** [comment]
@mayleighnmyers did you check if there are any other charts where .local is hardcoded?

**Comment by mayleighnmyers** [comment]
> @mayleighnmyers did you check if there are any other charts where .local is hardcoded?

@luksa I just changed the ones now. Do I need to change the version files that we no longer support?

**Comment by luksa** [comment]
To make this change in the older versions, you need to submit separate PRs in each branch. Those changes will then be pulled into 2.4 automatically.

**Review by luksa — APPROVED** [review]

**Comment by luksa** [comment]
@mayleighnmyers please make the same change in maistra-2.5. You'll have to do this manually, because cherry-picking won't work.

**Pull request merged**